apps/app/src/app/account/account-nua/account-nua.component.ts

```diff
@@ -16,57 +15,68 @@ import { TokenDialogComponent } from '../../components/token-dialog/token-dialog
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AccountNUAComponent {
-
-  @Input() zones: Zone[] | undefined;
-
   clients = this.nua.clients;
   expanded: string[] = [];
   unsubscribeAll = new Subject<void>();
 
   constructor(
     private nua: AccountNUAService,
     private dialog: MatDialog,
-    private cdr: ChangeDetectorRef
-  ) { 
+    private cdr: ChangeDetectorRef,
+    private router: Router,
+  ) {
     this.nua.updateClients();
   }
 
   askConfirmation(client: NUAClient): void {
-    this.dialog.open(STFConfirmComponent, {
-      width: '480px',
-      data: {
-        title: 'generic.alert',
-        message: 'account.token_generation_warning',
-        confirmText: 'account.new_code',
-        minWidthButtons: '110px'
-      }
-    }).afterClosed()
+    this.dialog
+      .open(STFConfirmComponent, {
+        width: '480px',
+        data: {
+          title: 'generic.alert',
+          message: 'account.token_generation_warning',
+          confirmText: 'account.new_code',
+          minWidthButtons: '110px',
+        },
+      })
+      .afterClosed()
       .pipe(
         takeUntil(this.unsubscribeAll),
         filter((result) => !!result),
-        switchMap(() => this.nua.renewClient(client.client_id))
+        switchMap(() => this.nua.renewClient(client.client_id)),
       )
       .subscribe(({ token }) => {
         this.showToken(token);
         this.nua.updateClients();
       });
   }
 
+  goToActivity(client: NUAClient): void {
+    const currentPath = location.pathname;
```

Comment:
This approach is a bit hacky, we have a navigation service to provide proper route path. Go that way:
```ts
accountUrl = this.stateService.account.pipe(
    filter((account) => !!account),
    map((account) => this.navigation.getAccountUrl(account!.slug)))
```